src/coap_session.c

```diff
@@ -457,15 +459,44 @@ coap_endpoint_get_session(coap_endpoint_t *endpoint,
       session->last_rx_tx = now;
       return session;
     }
-    if (session->ref == 0 && session->delayqueue == NULL && session->type == COAP_SESSION_TYPE_SERVER) {
+    if (session->ref == 0 && session->delayqueue == NULL &&
+        session->type == COAP_SESSION_TYPE_SERVER) {
       ++num_idle;
       if (oldest==NULL || session->last_rx_tx < oldest->last_rx_tx)
         oldest = session;
+
+      if (session->state == COAP_SESSION_STATE_HANDSHAKE) {
+        ++num_hs;
+        /* See if this is a partial SSL session set up
+           which needs to be cleared down to prevent DOS */
+        if ((session->last_rx_tx + 30 * COAP_TICKS_PER_SECOND) < now) {
```

Comment:
I would like to define the timeout as a pre-processor flag.

Reply:
Updated 30*COAP_TICKS_PER_SECOND with COAP_PARTIAL_SESSION_TIMEOUT_TICKS